### ISSUE TITLE
[BFCL] Bugs in decode executable predictions

### ISSUE THREAD

**Issue by XuHwang** [ShishirPatil/gorilla#652]
**Describe the issue**
BFCLv3 has modified the decode logic for executable predictions, while I found there are a lot of cases in REST category that could not be decoded by current method.

For example, here is one example:
```python
result="[requests.get(url=\"https://covid-193.p.rapidapi.com/statistics\", headers={\"X-RapidAPI-Key\": \"xxxxxxx\", \"X-RapidAPI-Host\": \"covid-193.p.rapidapi.com\"}, timeout=5, params={\"country\": \"China\"})]"
```

It would be parsed by [`decoded_output_to_execution_list`](https://github.com/ShishirPatil/gorilla/blob/70b6a4a2144597b1f99d1f4d3185d35d7ee532a4/berkeley-function-call-leaderboard/bfcl/model_handler/utils.py#L767) function. 

Then the nested value `{"X-RapidAPI-Key": "xxxxxxx", "X-RapidAPI-Host": \"covid-193.p.rapidapi.com"}` would  be parsed by [`parse_nested_value`](https://github.com/ShishirPatil/gorilla/blob/70b6a4a2144597b1f99d1f4d3185d35d7ee532a4/berkeley-function-call-leaderboard/bfcl/model_handler/utils.py#L749), which would raise error "`str` object has no attribute `.items()`"

In summary, this logic only consider the case that the nested value is a sub function while ignoring the truth that the argument type is just a dict.

**ID datapoint**
1. Datapoint / Model Handler permalink: 
2. Issue: 
2. Gorilla repo commit #: 

**What is the issue**

**Proposed Changes**

{
 'previous_datapoint':[], 
 'updated_datapoint':[]
}

**Additional context**
